ADR/0039-spdx-support.md

```diff
@@ -0,0 +1,151 @@
+
+| CycloneDX Attribute             | SPDX Attribute                                             |
+|---------------------------------|------------------------------------------------------------|
+| Formulations.components = [{}]  | Relationships = [{                                         |
+|                                 |  spdxElementId = <A-PACKAGE-ID>                            |
+|                                 |  relationshipType=BUILD_TOOL_OF                            |
+|                                 |  relatedSpdxElement=<ROOT-DOCUMENT-ID>                     |
+|                                 | }]                                                         |
+
+
+#### Metadata.tools
+The CycloneDX metadata.tools sub attributes that we are mostly interested in are the vendor and name elements. Information about the creation of the SPDX document can be stored into creationInfo. CreationInfo.creators element is basically a list of strings. There’s a vague specification ([here](https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#68-creator-field)]) about how it should be structured in the standard. Strings should be formatted in the following way: “<Attribute>: <Value>”. For example vendor should be stored as “Vendor: <vendor>”
+
+| CyloneDX Attribute                             | SPDX Attribute                                    |
+|------------------------------------------------|---------------------------------------------------|
+| Metadata.tools = [{“vendor”: “X”, “name”: “Y”] | CreationInfo.creators = [“Vendor: X”, “Tool: Y”]  |
+
+
+#### Merging SPDX
+##### Packages
+Packages of two SPDX documents can be merged together as a concatenation of two lists. In cycloneDX component elements can have only a single purl attribute, therefore component elements representing packages with the same name and version but with different purl have to be stored as multiple elements. SPDX package elements can bear multiple purls. Therefore multiple cycloneDX components can be squashed together into single SPDX package element with purls concatenated into a single list. Following rules are applied to packages merging process:
+- Packages with the same purl package name and version and type are squashed into single package element
+NOTE: packages cannot be merged together based on SPDXID attribute as there’s no specification in the spdx standard on how SPDXID should be calculated. Individual tools can calculate it differently while still passing condition to make it unique across the whole document.
+
+##### Relationships
+SPDX relationships represent graph/tree structure of relations of elements in the document. The Root element is the SPDX document itself (with SPDXID SPDXRef-Document). Individual packages are in relationship CONTAINS with the root document. In the case of packages which were used to build the container, packages are in relationship BUILD_TOOL_OF with the root document.
```

Comment:
> packages are in relationship ...

Does this mean that a SBOM describing a container would contain not just the information about packages within it, but also about everything used to build it?

Reply:
It depends. If the relationship is DEPENDS_ON then it match normal `Requires` from RPM spec. If the relationship is BUILD_DEPENDENCY_OF then it matches 'BuildRequires'  and then yes - it would contain information about packages used to build it. This may be required for SBOMs of type `Build`. For more information about relationship types see https://spdx.github.io/spdx-spec/v2.3/relationships-between-SPDX-elements/ For SBOM types see https://www.cisa.gov/sites/default/files/2023-04/sbom-types-document-508c.pdf